### PR TITLE
Significantly reduce memory leaks in anal tests

### PR DESCRIPTION
**Pull request by Lazula**
**Description**

Fix lots of memory leaks for anal tests. Not totally clean yet.

### CODE REVIEW

**Review by trufae — CHANGES_REQUESTED** [review]
you cant call rbin apis from shlr/

File: libr/anal/class.c [hunk 1]
```diff
@@ -599,7 +639,9 @@ static void r_anal_class_method_delete_class(RAnal *anal, const char *class_name
 	}
 	char *cur;
 	sdb_aforeach (cur, array) {
-		r_anal_class_unset_flag (anal, flagname_method (class_name, cur));
+		char *flagname = flagname_method (class_name, cur);
```
[the remark this answers is not in the document]
Comment (Lazula) [reply libr/anal/class.c:642]: what do you mean?

File: libr/asm/p/asm_dalvik.c [hunk 2]
```diff
@@ -525,6 +525,7 @@ static int dalvik_disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
 		op->size = size + op->payload;
 	}
 	free (strasm);
+	R_FREE (flag_str);
```
[the remark this answers is not in the document]
Comment (Lazula) [reply libr/asm/p/asm_dalvik.c:528]: i used R_FREE for the autocast from const, will change

File: libr/anal/class.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lazula) [reply libr/anal/class.c:642]: unset_flag does have null checking, and its done like this elsewhere

**Comment by swoops** [comment]
Nice work! You caught a few leaks I had been hunting.

**Pull request merged**